### PR TITLE
Moving files from sub projects

### PR DESCRIPTION
**Pull request by maxinelasp**
# Change Summary

## Overview
Moved the project templates up from sds-data-manager and imap-processing. I'm not sure if all these templates belong here, I'd be happy to hear thoughts.
I also converted the bug reporting and new feature templates to yaml, mostly for my own curiosity and to investigate. I'd like to suggest that future templates use yaml instead of markdown, although they seem pretty similar. It's not to hard to convert them. 
Finally, I added "IMAP" as a project on all of these templates for tracking purposes.

## New Dependencies
None

## New Files

- algorithm-development.md
   - Moved up from imap-processing, added IMAP project, and renamed. This one might just need to stay in imap-processing.

- bug-reporting.yml
   -  Moved up from imap-processing, converted to yaml, added IMAP project
   
- general-issue.md
    -  Moved up from imap-processing. This file might need to be deleted, it serves a similar purpose as new-feature
- new-feature.yml
    - Moved up from sds-data-manager, converted to yaml, added IMAP project
- spike-template.md
   - Moved up from sds-data-manager, added IMAP project

## Updated Files
- renamed L5_requirement.yml to l5-requirement.yml to match name scheme

## Testing

N/A

### CODE REVIEW

**Comment by maxineofficial** [comment]
The text of these templates didn't change at all, so you don't need to review that part!

**Comment by greglucas** [comment]
It looks like GitHub doesn't completely like the yaml. Specifically it doesn't seem to like empty strings.
https://github.com/IMAP-Science-Operations-Center/.github/blob/096b3404908cb6f3aad7833f82c49c2ef9c6e668/.github/ISSUE_TEMPLATE/new-feature.yml

I agree with using yaml because that is what GitHub seems to prefer: https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository
Do you want to convert all of the others over in this PR as well since you have a mix currently?

**Comment by maxineofficial** [comment]
@greglucas I will fix those errors with another commit. I can convert the others to yaml as well, but I'd like to make sure we want all of the templates first, because there are some duplicates.

**Comment by maxineofficial** [comment]
@GFMoraga might be interested this review since they did the initial templates

File: .github/ISSUE_TEMPLATE/bug-reporting.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+title: 'BUG -'
+labels: bug
+projects: IMAP
+body:
+- type: textarea
+  attributes:
+    label: "## Description of the issue"
+- type: textarea
+  attributes:
+    label: "### Steps to reproduce the issue
```
Comment (greglucas) [inline .github/ISSUE_TEMPLATE/bug-reporting.yml]:
```suggestion
    label: "Steps to reproduce the issue
```

In the yaml the "###" gets rendered with the title, so we should remove all of the "#"'s from these files.

**Review by greglucas — APPROVED** [review]
One minor comment, but this looks good to me. I think you can specify per repository what you want to bring in / how to display them too.

**Pull request merged**